packages/ra-ui-materialui/src/input/NullableBooleanInput.tsx

```diff
@@ -65,6 +66,21 @@ const NullableBooleanInput: FunctionComponent<
         source,
         validate,
     });
+
+    const enhancedOptions = displayNull
```

Comment:
there is something I don't really like with your syntax: `displayNull` takes precedence over `options`. 

I'd do it differently: prepare a static options object outside of the component:

```js
const displayNullOptions = { SelectProps: { displayEmpty: true }, InputLabelProps: { shrink: true } }
```

Then pass it ti TextField if `displayNull` is true:

```jsx
<TextField
   {displayNull ? ...displayNullOptions : ...{} }
   {...options}
```

Reply:
I would like this case to work fine : 
```
<NullableBooleanInput
   source="source"
   displayNull
   options={{SelectProps: { anotherCustomProp: true }}}
/>
```

I think, with your syntax, that the 
`SelectProps: { displayEmpty: true }`
is totally overrided by the 
`SelectProps: { anotherCustomProp: true }`

My code wanted to mix them in
```
SelectProps: { 
   displayEmpty: true,
   anotherCustomProp: true,
}```

Reply:
I changed for options to precede on displayNull, without erasing it